### ISSUE TITLE
How to clear site permissions data?

### ISSUE THREAD

**Issue by aquinn39** [MicrosoftEdge/WebView2Feedback#2889]
Calling CoreWebView2.Profile.ClearBrowsingDataAsync() seems to reset all permission data for websites, however, it also clears all other data for sites. I would like to clear just permission data. I have checked https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2browsingdatakinds?view=webview2-dotnet-1.0.1370.28, but none of the fields mention permissions at all.
Essentially, I'm having a similar issue to theses:
• https://github.com/MicrosoftEdge/WebView2Feedback/issues/1572
• https://github.com/MicrosoftEdge/WebView2Feedback/issues/2672

So basically I would like to know how we can reset site permissions for WebView2 without resetting other data. In the Edge Legacy WebView, I'm pretty sure it doesn't even remember permissions and so the permission requested event is raised every time a permission is requested and it's up to the app to remember the user's preferences if it wants; therefore, the Edge Legacy WebView does not have this issue.

**Comment by aquinn39** [comment]
Actually, it seems that CoreWebView2.Profile.ClearBrowsingDataAsync() does not clear permission data.

**Comment by PrakashC1** [comment]
Actually I want to prompt the browser permission dialog whenever I need(**Note: even I blocked that permission**). But once I blocked the permission, i am not able to re-prompt it. So I thought to clear site permissions. But now it also looks like not possible

Could you please provide any solution.........

**Comment by vickiez** [comment]
Hi @aquinn39 and @PrakashC1, we added an experimental [SavesInProfile](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2permissionrequestedeventargs.savesinprofile?view=webview2-dotnet-1.0.1619-prerelease#microsoft-web-webview2-core-corewebview2permissionrequestedeventargs-savesinprofile) property to PermissionRequestedEventArgs. You can set this to false to prevent the state from being saved and continue to get PermissionRequested events. Does this fix your issue? 

We've also added experimental APIs [GetNonDefaultPermissionSettingsAsync ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2profile.getnondefaultpermissionsettingsasync?view=webview2-dotnet-1.0.1619-prerelease#microsoft-web-webview2-core-corewebview2profile-getnondefaultpermissionsettingsasync) and [SetPermissionState](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2profile.setpermissionstateasync?view=webview2-dotnet-1.0.1619-prerelease#microsoft-web-webview2-core-corewebview2profile-setpermissionstateasync(microsoft-web-webview2-core-corewebview2permissionkind-system-string-microsoft-web-webview2-core-corewebview2permissionstate)) that may be helpful.

**Comment by vickiez** [comment]
@aquinn39  and @PrakashC1 the extended permission APIs are now available in the latest release package, 1.0.1661.34. Thanks!
